ql/src/java/org/apache/hadoop/hive/llap/ProactiveEviction.java

```diff
@@ -229,12 +230,59 @@ public String getSingleDbName() {
     }
 
     /**
-     * Match a CacheTag to this eviction request.
+     * Match a CacheTag to this eviction request. Must only be used on LLAP side only, where the received request may
```

Comment:
I do not get this comment. "request may only contain one information for one DB".
Could you please ellaborate?

Reply:
Proactive eviction requests are on a per DB basis: https://github.com/apache/hive/blob/master/llap-common/src/protobuf/LlapDaemonProtocol.proto#L234-L238
This comment is just a heads-up of that fact.

Reply:
Why is this checked here, and not at construction time, or when we receive / parse the request? This seems a little awkward for me.

Reply:
This Request class is used on HS2 side too, and there it may contain multiple DBs, hence there is no such 'general' restriction in this class. Although we're not doing this right now, but It's not impossible to call this method while we have multiple DBs, that's why the javadoc details the restriction on this method only.

Reply:
Thanks for the clarification!